src/components/views/right_panel/TimelineCard.tsx

```diff
@@ -221,6 +221,9 @@ export default class TimelineCard extends React.Component<IProps, IState> {
 
         const isUploading = ContentMessages.sharedInstance().getCurrentUploads(this.props.composerRelation).length > 0;
 
+        const myMembership = this.props.room.getMyMembership();
+        const canSpeak = myMembership === "join";
```

Comment:
This should be using something akin to https://github.com/matrix-org/matrix-react-sdk/blob/21d3cb08a9223cac99e1b6bce42090a828256135/src/components/views/rooms/MessageComposer.tsx#L368 to account also for permissions & tombstones.

Reply:
Ah okay, I just copied from:

 https://github.com/matrix-org/matrix-react-sdk/blob/44776a1c5c81209abdad60b1766a3a963d0980a0/src/components/structures/RoomView.tsx#L1986-L1989

Should it be changed as well?

Reply:
Will rename both my version and `RoomView`'s to `showComposer`, as that's what it really means (discussed [here](https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$9W3no-bx8A5SmFEqNYum3jA_AXt8lwU2isAtpQ9gbTg?via=matrix.org&via=element.io&via=vector.modular.im).